### PR TITLE
vscode-extensions.vadimcn.vscode-lldb: remove custom lldb

### PR DESCRIPTION
**Pull request by schuelermine**
## Description of changes

Remove custom LLDB from `vscode-extensions.vadimcn.vscode-lldb`. It is unnecessary (Rust-specific features still appear on my end), the repository seems abandoned, and it needs a patch for an issue which seems to be fixed except for here.
Also removes the broken update script.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Congee** [comment]
Result of `nixpkgs-review pr 246652` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>vscode-extensions.vadimcn.vscode-lldb</li>
  </ul>
</details>

**Comment by Congee** [comment]
> Result of `nixpkgs-review pr 246652` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 1 package failed to build:

Actually the failure happens prior to this PR, too. I'd love to get rid of the custom lldb. It's not friendly to maintainers, IMO

**Comment by schuelermine** [comment]
Hmm, I will check again. I think it built on the commit I based the branch off?

**Comment by Congee** [comment]
> Hmm, I will check again. I think it built on the commit I based the branch off?

Fine. The build failure happens on aarch64-darwin. It builds on my x86_64-linux.

**Review by Congee — APPROVED** [review]

**Comment by schuelermine** [comment]
Oh sorry, I missed that you checked on Darwin/ARM.

File: pkgs/applications/editors/vscode/extensions/vadimcn.vscode-lldb/update.sh
Comment (meskill) [inline pkgs/applications/editors/vscode/extensions/vadimcn.vscode-lldb/update.sh:1]: `default.nix` still points out to `update.sh` in `passthru`

Comment (schuelermine) [reply pkgs/applications/editors/vscode/extensions/vadimcn.vscode-lldb/update.sh:1]: Done!

**Comment by meskill** [comment]
I've tried to build this version and this is definitely more restricted version related to Rust features than custom lldb.

I found out this after reading [the discussion](https://github.com/vadimcn/codelldb/discussions/699) and you can find the examples below

<details>
<summary>Gifs with demonstration</summary>

Here as it works before:
Before

and after removing custom lldb:
After

</details>

**Comment by schuelermine** [comment]
Oh. In that case I’ll close this.

**Pull request closed without merging**